### PR TITLE
Add clang-format script

### LINKED ISSUE #3215: Reformat all code and add `clang-format` CI job
Currently PRs and commits aren't well-checked if they have proper formatting. In analogy with other projects under Rizinorg reformat it with `clang-format` and add corresponding CI job.

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

Added script like in Rizin to automatically reformat all files at once.

**Test plan (required)**

CI is green

**Closing issues**

Closes https://github.com/rizinorg/cutter/issues/3215

### CODE REVIEW

File: src/Main.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * @brief Attempt to connect to a parent console and configure outputs.
  */
 #ifdef Q_OS_WIN
-#include <windows.h>
+#    include <windows.h>
```
Comment (ret2libc) [inline src/Main.cpp:16]: do we do this in Rizin??

**Review by ret2libc — COMMENTED** [review]
I'm ok with adding the script.

File: src/Main.cpp (hunk 1 above)
Comment (XVilka) [reply src/Main.cpp:16]: No idea, but the script uses already existing `_clang-format` file from Cutter repository: https://github.com/rizinorg/cutter/blob/dev/_clang-format

**Review by Rot127 — APPROVED** [review]
Don't see any reason to not do it.

File: scripts/clang-format.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import sys
+
+from git import Repo
+
+dirlist = [
+    "src",
+]
+
+skiplist = [
+    "translations",
```
Comment (wargio) [inline scripts/clang-format.py:19]: skip rizin

File: scripts/clang-format.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+]
+
+patterns = ["*.c", "*.cpp", "*.h", "*.hpp", "*.inc"]
```
Comment (wargio) [inline scripts/clang-format.py]: only parse `.cpp` and `.h`

File: scripts/clang-format.py (hunk 2 above)
Comment (XVilka) [reply scripts/clang-format.py:19]: No need, it's on another level

**Review by wargio — APPROVED** [review]

**Pull request merged**